central/complianceoperator/v2/scanconfigurations/service/service_impl.go

```diff
@@ -392,6 +393,53 @@ func (s *serviceImpl) GetMyReportHistory(ctx context.Context, request *v2.Compli
 	return res, nil
 }
 
+func (s *serviceImpl) DeleteReport(ctx context.Context, req *v2.ResourceByID) (*v2.Empty, error) {
+	if req.GetId() == "" {
+		return nil, errors.Wrap(errox.InvalidArgs, "Report Snapshot ID is required for deletion")
+	}
+
+	slimUser := authn.UserFromContext(ctx)
+	if slimUser == nil {
+		return nil, errors.New("Could not determine user identity from provided context")
+	}
+
+	snapshot, found, err := s.snapshotDS.GetSnapshot(ctx, req.GetId())
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to retrieve Report Snapshot %s", req.GetId())
+	}
+	if !found {
+		return nil, errors.Wrapf(errox.NotFound, "Unable to find the Report Snapshots %s", req.GetId())
+	}
+
+	if slimUser.GetId() != snapshot.GetUser().GetId() {
+		return nil, errors.Errorf("The user %s cannot delete the report %s", slimUser.GetId(), snapshot.GetReportId())
+	}
```

Comment:
Potential security issue (as in #13008), where user without permission can check whether a snapshot exists or not.
A solution here would be to generate identical error message for both cases and add logging so that we can distinguish those two situations.

Reply:
Let me give it a thought. I think it's a valid concern.